### ISSUE TITLE
Why ExcludeAdvice is sealed?

### ISSUE THREAD

**Issue by Kahbazi** [ArxOne/MrAdvice#87]
I have two projects, let's call them A and B.

in project A , I have one methodAdvice 

`    public class LogAdvice : Attribute, IMethodAdvice`

in project B , I set this attribute to assembly in order to advice all of my classes.
Although I need to exclude some classes from advising , and I don't want to use ExcludePointcut.

what I have in my mind is to use ExcludeAdvice Attribute which should be provided in project A and use it in project B.

I want to write this in project A : 

```
public class ExcludeLogAdvicesAttribute : ExcludeAdvicesAttribute
{
 // in constructor I set value for advicesTypes
}
```

now in project B I can easily use [ExcludeLogAdvicesAttribute] without having to know it's adviseType, and if I change my advice name, I only need to change it once in project A.

To sum up, in order to do that I need for ExcludeAdvicesAttribute to not be sealed.

So can in be not sealed?

**Comment by picrap** [comment]
You made the point. I sealed it because adding extra features to it wouldn't have been taken into account by the weaver.
But I understand your reasons, so it's OK to unseal it.

**Comment by picrap** [comment]
I was going to unseal this, but this is much more complicated that expected, since the Mr Advice weaver reads and interprets constructors directly, so there is no way to guess what an inherited class would do.
I keep this issue open to say that I'm still thinking about it, but this may be impossible to achieve, I'm afraid.
